### PR TITLE
i18n misc

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: modules/@angular/compiler/src/i18n/extractor_merger.ts [hunk 1]
```diff
@@ -204,28 +205,32 @@ class _Visitor implements html.Visitor {
...
     this._depth++;
     const wasInI18nNode = this._inI18nNode;
+    const wasInImplicitNode = this._inImplicitNode;
     let childNodes: html.Node[];
 
     // Extract only top level nodes with the (implicit) "i18n" attribute if not in a block or an ICU
     // message
     const i18nAttr = _getI18nAttr(el);
-    const isImplicitI18n = this._implicitTags.some((tag: string): boolean => el.name === tag);
+    const isImplicit = this._implicitTags.some((tag: string): boolean => el.name === tag) &&
```
Comment (vsavkin) [inline modules/@angular/compiler/src/i18n/extractor_merger.ts]: I don't think you need to type the callback. At least typescript 2.0, and I'm pretty sure 1.8 will infer the types.

File: modules/@angular/compiler/test/i18n/digest_spec.ts [hunk 2]
```diff
@@ -7,32 +7,52 @@
...
+      //   while (result.length < size) {
+      //     result += result;
+      //   }
+      //   result = result.slice(-size);
+      // }
+      //
+      // console.log(sha1(result));
+      const prefix = `你好，世界`;
+      let result = sha1(prefix);
+      for (let size = prefix.length; size < 5000; size += 101) {
```
Comment (vsavkin) [inline modules/@angular/compiler/test/i18n/digest_spec.ts]: Why do we need to have this loop in the test? Can't we just hardcode the value?

File: modules/@angular/compiler/src/i18n/extractor_merger.ts (hunk 1 above)
Comment (vicb) [inline modules/@angular/compiler/src/i18n/extractor_merger.ts]: yep should work

**Comment by alxhub** [comment]
Merged to master as b65f66fe.

**Pull request closed without merging**